ansible/inventory/multi_inventory.py

```diff
@@ -119,144 +227,103 @@ def load_yaml_config(self, conf_file=None):
 
         return config
 
-    def get_provider_tags(self, provider, env=None):
-        """Call <provider> and query all of the tags that are usuable
-        by ansible.  If environment is empty use the default env.
-        """
-        if not env:
-            env = os.environ
-
-        # Allow relatively path'd providers in config file
-        if os.path.isfile(os.path.join(self.file_path, provider)):
-            provider = os.path.join(self.file_path, provider)
-
-        # check to see if provider exists
-        if not os.path.isfile(provider) or not os.access(provider, os.X_OK):
-            raise RuntimeError("Problem with the provider.  Please check path " \
-                        "and that it is executable. (%s)" % provider)
-
-        cmds = [provider]
-        if self.args.get('host', None):
-            cmds.append("--host")
-            cmds.append(self.args.get('host', None))
-        else:
-            cmds.append('--list')
-
-        if 'aws' in provider.lower():
-            cmds.append('--refresh-cache')
-
-        return subprocess.Popen(cmds, stderr=subprocess.PIPE, \
-                                stdout=subprocess.PIPE, env=env)
-
-    @staticmethod
-    def generate_config(provider_files):
-        """Generate the provider_files in a temporary directory.
-        """
-        prefix = 'multi_inventory.'
-        tmp_dir_path = tempfile.mkdtemp(prefix=prefix)
-        for provider_file in provider_files:
-            filedes = open(os.path.join(tmp_dir_path, provider_file['name']), 'w+')
-            content = Template(provider_file['contents']).substitute(tmpdir=tmp_dir_path)
-            filedes.write(content)
-            filedes.close()
-
-        return tmp_dir_path
-
-    def run_provider(self):
-        '''Setup the provider call with proper variables
-           and call self.get_provider_tags.
-        '''
+    def run_account_providers(self, force=False, valid_cache=True, account=None):
+        """Setup the provider call with proper variables and call self.get_provider_tags"""
```

Comment:
I don't see how you're handling the case where `account == None`.

What is expected to happen in that case?

`account` can be `None` if:
* They fail to pass it into this function
* A clusterid is specified, but doesn't exist in an account
* probably others